sonic_db_config/db_config.go

```diff
@@ -1,135 +1,237 @@
-        DbInit()
-    }
-    db, ok := sonic_db_config["DATABASES"].(map[string]interface{})[db_name]
-    if !ok {
-        panic(fmt.Errorf("database name '%v' is not valid in database_config.json file!", db_name))
-    }
-    inst_name, ok := db.(map[string]interface{})["instance"]
-    if !ok {
-        panic(fmt.Errorf("'instance' is not a valid field in database_config.json file!"))
-    }
-    inst, ok := sonic_db_config["INSTANCES"].(map[string]interface{})[inst_name.(string)]
-    if !ok {
-        panic(fmt.Errorf("instance name '%v' is not valid in database_config.json file!", inst_name))
-    }
-    return inst.(map[string]interface{})
-}
-
-func GetDbSeparator(db_name string)(string) {
-    if !sonic_db_init {
-        DbInit()
-    }
-    db_list := GetDbList()
-    separator, ok := db_list[db_name].(map[string]interface{})["separator"]
-    if !ok {
-        panic(fmt.Errorf("'separator' is not a valid field in database_config.json file!"))
-    }
-    return separator.(string)
-}
-
-func GetDbId(db_name string)(int) {
-    if !sonic_db_init {
-        DbInit()
-    }
-    db_list := GetDbList()
-    id, ok := db_list[db_name].(map[string]interface{})["id"]
-    if !ok {
-        panic(fmt.Errorf("'id' is not a valid field in database_config.json file!"))
-    }
-    return int(id.(float64))
-}
-
-func GetDbSock(db_name string)(string) {
-    if !sonic_db_init {
-        DbInit()
-    }
-    inst := GetDbInst(db_name)
-    unix_socket_path, ok := inst["unix_socket_path"]
-    if !ok {
-        panic(fmt.Errorf("'unix_socket_path' is not a valid field in database_config.json file!"))
-    }
-    return unix_socket_path.(string)
-}
-
-func GetDbHostName(db_name string)(string) {
-    if !sonic_db_init {
-        DbInit()
-    }
-    inst := GetDbInst(db_name)
-    hostname, ok := inst["hostname"]
-    if !ok {
-        panic(fmt.Errorf("'hostname' is not a valid field in database_config.json file!"))
-    }
-    return hostname.(string)
-}
-
-func GetDbPort(db_name string)(int) {
-    if !sonic_db_init {
-        DbInit()
-    }
-    inst := GetDbInst(db_name)
-    port, ok := inst["port"]
-    if !ok {
-        panic(fmt.Errorf("'port' is not a valid field in database_config.json file!"))
-    }
-    return int(port.(float64))
-}
-
-func GetDbTcpAddr(db_name string)(string) {
-    if !sonic_db_init {
-        DbInit()
-    }
-    hostname := GetDbHostName(db_name)
-    port := GetDbPort(db_name)
-    return hostname + ":" + strconv.Itoa(port)
+func GetDbDefaultNamespace() string {
+	return SONIC_DEFAULT_NAMESPACE
+}
+func CheckDbMultiNamespace() bool {
+	if !sonic_db_init {
+		DbInit()
+	}
+	return sonic_db_multi_namespace
+}
+func GetDbNonDefaultNamespaces() ([]string, int) {
+	if !sonic_db_init {
+		DbInit()
+	}
+	ns_list := make([]string, 0, len(sonic_db_config))
+	for ns := range sonic_db_config {
+		if ns == SONIC_DEFAULT_NAMESPACE {
+			continue
+		}
+		ns_list = append(ns_list, ns)
+	}
+	return ns_list, len(ns_list)
```

Comment:
1) `len` is golang builtin function, that could be called on array/string without overhead and it's a well known way to get elements count with `len`
2) the result is not used after calls of this function
So as it is a builtin and it is ignored it is redundant

Reply:
updated not to return length as it is not used.